applications/conceptual-model-editor/src/action/create-profile-of-selection.ts

```diff
@@ -0,0 +1,116 @@
+import { isSemanticModelClassUsage, isSemanticModelRelationshipUsage, SemanticModelRelationshipEndUsage } from "@dataspecer/core-v2/semantic-model/usage/concepts";
+import { createClassProfile } from "./open-create-profile-dialog";
+import { findSourceModelOfEntity } from "../service/model-service";
+import { InMemorySemanticModel } from "@dataspecer/core-v2/semantic-model/in-memory";
+import { createRelationshipUsage } from "@dataspecer/core-v2/semantic-model/usage/operations";
+import { addSemanticClassProfileToVisualModelAction } from "./add-class-profile-to-visual-model";
+import { UseDiagramType } from "../diagram/diagram-hook";
+import { addSemanticRelationshipProfileToVisualModelAction } from "./add-relationship-profile-to-visual-model";
+
+
+export const profileSelectionAction = (nodeSelection: string[], edgeSelection: string[], classesContext: ClassesContextType, graph: ModelGraphContextType,
+                                      visualModel: VisualModel | null, diagram: UseDiagramType, notifications: UseNotificationServiceWriterType, options: Options): void => {
+  const createdClassesAndClassesProfiles: Record<string, string> = {};
+  for(const selectedEntityId of nodeSelection) {
+    const classOrClassProfileToBeProfiled = graph.aggregatorView.getEntities()?.[selectedEntityId]?.aggregatedEntity;
+    if(classOrClassProfileToBeProfiled === undefined) {
+      notifications.error("The entity (node) to be profiled from selection is not present in aggregatorView");
+      return;
+    }
+    const isTheProfiledClassClassProfile = isSemanticModelClassUsage(classOrClassProfileToBeProfiled);
+    if(!isSemanticModelClass(classOrClassProfileToBeProfiled) && !isTheProfiledClassClassProfile) {
+      notifications.error("The entity to be profiled from selection is not a class or class profile");
+      return;
+    }
+
+    const profileClassState = createCreateProfileClassDialogState(
+      classesContext,
+      graph,
+      visualModel,
+      options.language,
+      classOrClassProfileToBeProfiled,
+    );
+    const createdClassProfile = createClassProfile(profileClassState);
+    if(createdClassProfile === null) {
+      continue;
+    }
+
+    createdClassesAndClassesProfiles[selectedEntityId] = createdClassProfile.identifier;
+
+    if(isWritableVisualModel(visualModel)) {
+      addSemanticClassProfileToVisualModelAction(notifications, graph, visualModel, diagram, createdClassProfile.identifier, createdClassProfile.model.getId(), null);
+    }
+  }
+
+  for(const selectedEntityId of edgeSelection) {
+    const associationOrAssociationProfileToBeProfiled = graph.aggregatorView.getEntities()?.[selectedEntityId]?.aggregatedEntity;
+    if(associationOrAssociationProfileToBeProfiled === undefined || associationOrAssociationProfileToBeProfiled === null) {
+      notifications.error("The entity (edge) to be profiled from selection is not present in aggregatorView");
+      continue;
```

Comment:
So it is better to fail of the elements if there is some faulty one? I guess that it makes sense, since this is most likely programmer error, if it occurrs.